rules/do-you-know-how-modern-stateless-auth-works/rule.md

```diff
@@ -0,0 +1,75 @@
+---
+type: rule
+title: Do you know how Modern Stateless Authentication works?
+uri: modern-stateless-authentication
+authors:
+  - title: Dhruv Mathur
+    url: https://www.ssw.com.au/people/dhruv-mathur/
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming/
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/matt-goldman/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+# Do you know how Modern Stateless Authentication works?
+
+Modern stateless authentication, particularly through JSON Web Tokens (JWT), is a widely adopted method in web development for managing user sessions and securing applications. Unlike traditional session-based authentication, JWT allows for a more scalable and efficient approach by avoiding server-side storage of user session information.
+
+This new approach to authentication often utilizes the OpenID Connect (OIDC) authentication fraemwork, which is an authentication layer built on top of OAuth 2.0. The key role of OIDC is to unify the way in which authentication services and their users communicate, ensuring a consistent and standardized interaction
+
+A JWT is a base64-encoded JSON object that is digitally signed and optionally encrypted. It consists of three parts: a header, a payload, and a signature. The header contains metadata about the token, such as the type and the signing algorithm. The payload contains claims, which are statements about the user and additional data. The signature is used to verify the integrity of the token and to ensure that it has not been tampered with and also to establish that the token has been issued by a trusted authority.
+
+The process of stateless authentication usually involves trusting a ExternalAuthProvider outside your application to authenticate users and then using a token issued by them to identify the user in your application. An ExternalAuthProvider can be a service like Azure AD, IdentityServer, or any other OAuth2 or OpenID Connect provider.
+
+<https://www.youtube.com/embed/UBUNrFtufWo>
+**Figure: Session vs Token Authentication (2min)**
+<!--endintro-->
```

Comment:
Too much stuff in this intro... we usually have 1 or 2 paragraphs

- [ ] 1. Please move this divisor up

Reply:
Done - made the intro small and concise.